### DISCUSSION TITLE
How do you throw errors from a synchronous method?

### DISCUSSION THREAD

**Discussion by itsramiel** [mateusz1913/rnbridgingtutorial#58] [Q&A] [answered]
Hi,

how do you throw errors from synchronous methods? Does using the platform exceptions work?

**Comment by mateusz1913** [comment, marked as answer]
Hello @itsramiel, not sure to be honest, you can try to raise an exception and maybe it will be redirected to JS layer. However, I'd rather advise to not do it and just return some result/optional value instead, so that everything can be handled gracefully in JS layer

**Reply by itsramiel** [reply]
Thank you for the answer. Yeah I tried to throw a RuntimeException from kotlin and you get the redbox but you cant catch it with try/catch.

I am going with as you said returning a value from native and handling throwing the error from ks. Thanks for the reply. I wanted to make sure whether there is a better way.
